app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { fetchRandomMessage } from '../actions';
+
+const Greeting = ({ randomMessage, fetchRandomMessage }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [countdown, setCountdown] = useState(10);
+
+  useEffect(() => {
+     const fetchMessage = async () => {
+      try {
+        await fetchRandomMessage();
+        setIsLoading(false);
+      } catch (error) {
+        throw new Error('Error fetching data:', error);
+      }
+    };
+
+    fetchMessage();
+  }, [fetchRandomMessage]);
+
+  useEffect(() => {
+    if (countdown === 0) {
+      fetchRandomMessage();
+      setCountdown(10);
+    } else {
+      const timer = setTimeout(() => setCountdown(countdown - 1), 1000);
+      return () => clearTimeout(timer);
+    }
+  }, [countdown, fetchRandomMessage]);
+
+  if (isLoading) {
+    return <div class="loading">Loading...</div>;
+  }
+
+    return (
+       <div class="greeting-content">
+         <h1>Greeting of the day</h1>
+         <p>"{randomMessage}"</p>
+         <h2>Greeting changes in: <span>{countdown}</span></h2>
+       </div>
+     );
+   };
+
+   const mapStateToProps = (state) => {
+     return {
+       randomMessage: state.randomMessage,
+     };
+   };
+
+   export default connect(mapStateToProps, { fetchRandomMessage })(Greeting);
```

Comment:
**OPTIONAL** - You have done well so far, but since you are working with functional components and not class components, it is advised to use the `useSelector` hook to fetch the app's state from redux. Please fix this 🙏 
**OPTIONAL** - Also, please make use of the `useDispatch` hook to dispatch your API calls.